static/app/components/core/alert.chonk.tsx

```diff
@@ -0,0 +1,130 @@
+import type {AlertProps} from 'sentry/components/alert';
+import {chonkStyled, useChonkTheme} from 'sentry/utils/theme/theme.chonk';
+import type {ChonkPropMapping} from 'sentry/utils/theme/withChonk';
+
+const chonkAlertPropMapping: ChonkPropMapping<AlertProps, ChonkAlertProps> = props => {
+  return {
+    ...props,
+    type:
+      props.type === 'muted' ? 'subtle' : props.type === 'error' ? 'danger' : props.type,
+  };
+};
+
+interface ChonkAlertProps extends Omit<AlertProps, 'type'> {
+  type: 'subtle' | 'info' | 'warning' | 'success' | 'danger';
+  size?: 'sm';
+}
+
+function AlertPanel({type, size, ...props}: ChonkAlertProps) {
+  const theme = useChonkTheme();
+
+  return (
+    <AlertPanelDiv
+      {...props}
+      colors={chonkAlertTheme(type, theme)}
+      padding={chonkAlertPadding(size, theme)}
+    >
+      {props.children}
+    </AlertPanelDiv>
+  );
+}
+
+const AlertPanelDiv = chonkStyled('div')<
+  Omit<ChonkAlertProps, 'type'> & {
+    colors: ReturnType<typeof chonkAlertTheme>;
+    padding: ReturnType<typeof chonkAlertPadding>;
+  }
+>`
+  background-color: ${p => p.colors.background};
+  color: ${p => p.colors.color};
+  border: ${p => p.colors.border};
```

Comment:
another idea I just had that could help us get rid of the `AlertPanel` wrapper would be to just call the `chonkAlertTheme` function inside the styled div. Since we now have `chonkStyled`, it would get the correctly typed theme, too:

```
const AlertPanelDiv = chonkStyled('div')<Omit<ChonkAlertProps, 'type'>>`
  ${p => chonkAlertTheme(p.type, p.theme)}
`
```

we would need to type `chonkAlertTheme` to return something that `satisfies CssProperties`, so it would need to return `backgroundColor` instead of `background`, but I think it would be a good simplification nonetheless.

Reply:
I have thought of this too, and I like that approach too, as it would group the CSS theming into a one place and create separation from the rest of the rules